app/src/main/java/org/simple/clinic/newentry/form/DateOfBirthField.kt

```diff
@@ -0,0 +1,20 @@
+package org.simple.clinic.newentry.form
+
+import org.simple.clinic.newentry.form.ValidationError.InvalidDateFormat
+import org.simple.clinic.newentry.form.ValidationError.MissingValue
+import org.threeten.bp.LocalDate
+import org.threeten.bp.format.DateTimeParseException
+
+class DateOfBirthField : InputField<String>() {
+  override fun validate(value: String): List<ValidationError> {
+    if (value.isBlank()) return listOf(MissingValue)
+
+    try {
+      LocalDate.parse(value)
```

Comment:
My concern with doing this in this way is that we are tying the entered value (which will generally be entered by a user) to the date format which `LocalDate` expects, which is `yyyy-MM-dd` which may not be what user's are used to.

We should try to change this method to accept a `org.threeten.bp.format.DateTimeFormatter` instead so that we can decouple the format from the validation.

Reply:
Yes, that is happening in one of the following commits.